boards/arm/rpi_pico/doc/index.rst

```diff
@@ -83,7 +83,7 @@ hardware features:
      - :dtcompatible:`raspberrypi,pico-flash`
    * - UART (PIO)
      - :kconfig:option:`CONFIG_SERIAL`
-     - :dtcompatible:`raspberrypi,pico-uart-pio`
+     - :dtcompatible:`raspberrypi,pico-pio-uart`
```

Comment:
I think it's better to leave it as is.
The format [manufacturere],[device]-[implement-variant] is commonly used.

for example.
bindings/sensor/bosch,bme280-i2c.yaml
bindings/sensor/bosch,bme280-spi.yaml